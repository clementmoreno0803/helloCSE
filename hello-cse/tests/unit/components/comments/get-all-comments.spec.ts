import { describe, it, expect } from 'vitest';
import { shallowMount } from '@vue/test-utils';
import GetAllComments from "@/components/comments/get-all-comments.vue";
import { mockComments } from "../../fixtures/mockComments.mock";

const generateWrapper = () => {
  return shallowMount(GetAllComments, {
    props: {
      comments: mockComments,
    },
  })
}

describe('GetAllComments', () => {

  it('should render correctly', () => {
    const wrapper = generateWrapper()

    expect(wrapper.element).toMatchSnapshot();
    expect(wrapper.exists()).toBe(true);
  });

  it('should display the correct number of comments', () => {
    const wrapper = generateWrapper()

    const numberOfComments = wrapper.find('.get-all-comments__number-of-comments');
    expect(numberOfComments.text()).toBe(`${mockComments.length}`);
  });

  it('should render all comments', () => {
    const wrapper = generateWrapper()

    const renderedComments = wrapper.findAll('.get-all-comments__list-of-comments .v-list-item');
    expect(renderedComments.length).toBe(mockComments.length);
  });

  it('should calculate minutes since published correctly', () => {
    const wrapper = generateWrapper()

    const minutesSincePublished = wrapper.vm.getMinutesSincePublished(mockComments[0].dateCreation);
    expect(minutesSincePublished).toBe(5); // Should be 5 minutes

    const hoursSincePublished = wrapper.vm.getMinutesSincePublished(mockComments[1].dateCreation);
    expect(hoursSincePublished).toBe(2); // Should return hours for 2 hours ago
  });
});
