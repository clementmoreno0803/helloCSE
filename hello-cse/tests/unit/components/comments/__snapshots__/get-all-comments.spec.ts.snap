// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`GetAllComments > should render correctly 1`] = `
<div
  data-v-6c752411=""
>
  <div
    class="get-all-comments__title"
    data-v-6c752411=""
  >
    <h3
      data-v-6c752411=""
    >
      Commentaires:
    </h3>
    <span
      class="get-all-comments__number-of-comments"
      data-v-6c752411=""
    >
      2
    </span>
  </div>
  <v-list
    class="get-all-comments__list-of-comments"
    data-v-6c752411=""
    lines="two"
  >
    
    <v-list-item
      data-v-6c752411=""
      prepend-avatar="https://thispersondoesnotexist.com/"
    >
      <div
        class="get-all-comments__comment-part"
        data-v-6c752411=""
      >
        <div
          class="get-all-comments__name"
          data-v-6c752411=""
        >
          <h3
            data-v-6c752411=""
          >
            User1
          </h3>
          <p
            data-v-6c752411=""
          >
            <span
              data-v-6c752411=""
              style="color: orange;"
            >
              8
            </span>
             /10 
          </p>
        </div>
        <h4
          data-v-6c752411=""
        >
          Il y a 5 minutes
        </h4>
        <p
          data-v-6c752411=""
        >
          This is a comment.
        </p>
      </div>
    </v-list-item>
    <v-list-item
      data-v-6c752411=""
      prepend-avatar="https://thispersondoesnotexist.com/"
    >
      <div
        class="get-all-comments__comment-part"
        data-v-6c752411=""
      >
        <div
          class="get-all-comments__name"
          data-v-6c752411=""
        >
          <h3
            data-v-6c752411=""
          >
            User2
          </h3>
          <p
            data-v-6c752411=""
          >
            <span
              data-v-6c752411=""
              style="color: orange;"
            >
              9
            </span>
             /10 
          </p>
        </div>
        <h4
          data-v-6c752411=""
        >
          Il y a 2 minutes
        </h4>
        <p
          data-v-6c752411=""
        >
          This is another comment.
        </p>
      </div>
    </v-list-item>
    
  </v-list>
</div>
`;
